modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -639,7 +639,16 @@ public String toDefaultValue(Property p) {
                 return null;
             }
 
-            return String.format(pattern, getTypeDeclaration(ap.getItems()));
+            String typeDeclaration = getTypeDeclaration(ap.getItems());
+            Object java8obj = additionalProperties.get("java8");
+            if (java8obj != null) {
+                Boolean java8 = Boolean.valueOf(java8obj.toString());
+                if (java8 != null && java8) {
+                    typeDeclaration = "";
```

Comment:
@JFCote What about returning earlier here with `return String.format(pattern, "")`?

For the return below (line 651), we keep the original one `return String.format(pattern, getTypeDeclaration(ap.getItems()));` Would that work for both Play and RestEasy generator?

cc @daveish

Reply:
@wing328 I can change it if you want, I learn the old school way of not having too much `return` in a function (even if that was only required in the old days). Let me know if you want me to change it.

As for the line 651, it the exact same return as before. It's just that I have declared `typeDeclaration` online 642. We change it to "" only if java8 is set to true.